core/server/common/src/main/java/alluxio/master/StateLockManager.java

```diff
@@ -131,7 +133,13 @@ public void mastersStartedCallback() {
    */
   public LockResource lockShared() throws InterruptedException {
     if (LOG.isDebugEnabled()) {
-      LOG.debug("Thread-{} entered lockShared().", ThreadUtils.getCurrentThreadIdentifier());
+      SAMPLING_LOG.debug(String.format("Thread-%s entered lockShared().",
```

Comment:
This will add this string to a map in sampling logger.. Do you think that's okay?

Reply:
Woudl that be useful though? Sampling log will filter many instances of this log when there are  a lot of them in a short period of time. Since it is indexed by format string, multiple thread logging at the same time will result in only one entry being logged.

Reply:
That's the intention. I am not that interested in a background thread that is going in and out of this lock repeatedly which can happen a lot in certain scenarios. There is a case for reentrant locks but I don't think that is so extreme to the point where we could hit the lock limit of 60K+.

Reply:
I am not sure if I understand the purpose of this change. Do you mind putting some explanation or example in the description?

Reply:
The purpose of this change is to reduce the amount of logging in this debug log. The real bit of change is the one below this where it's reporting all the waiters and holders when the number of shared locks is too high.

Reply:
In that case, it makes more sense to just downlevel this log to trace instead of sampling it. Sampling log on locking information isn't very useful in that we actually care about each individual occurrence within a short period of time.

Reply:
I'll just not change this.

Reply:
Fixed.